authors/apps/authentication/tests/test_signup.py

```diff
@@ -0,0 +1,37 @@
+# Python and Django imports
+from rest_framework.views import status
+
+
+# local imports
+from .base_test import TestBaseCase
+
+
+class TestRegistration(TestBaseCase):
+    def test_valid_user_registration(self):
+        response = self.registration
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_no_username_registration(self):
+        del self.test_user['user']['username']
+        response = self.client.post(self.signup_url,
+                                    self.test_user,
+                                    format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(b'This field is required', response.content)
+
+    def test_no_email_registration(self):
+        del self.test_user['user']['email']
+        response = self.client.post(self.signup_url,
+                                    self.test_user,
+                                    format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(b'This field is required', response.content)
+
+    def test_no_password_registration(self):
+        del self.test_user['user']['password']
+        response = self.client.post(self.signup_url,
+                                    self.test_user,
+                                    format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn(b'This field is required', response.content)
```

Comment:
@JoshuaKodhe This chunk of code is duplicated in about 4 or 5 more places. We need to be very keen on maintainability, create a single function/method that does this, then call it in all the relevant areas to reduce code duplication